distributed/comm/ucx.py

```diff
@@ -103,14 +102,55 @@ def peer_address(self):
         return self._peer_addr
 
     async def write(self, msg, serializers=None, on_error="message"):
-        nbytes = sys.getsizeof(msg)
-        await self.ep.send_msg(msg, nbytes)
-        return nbytes
+        frames = await to_frames(msg,
+                                 serializers=serializers,
+                                 on_error=on_error)  # TODO: context=
+
+        msg = b''.join(frames)
+        # nbytes = sys.getsizeof(msg)
+
+        # Do we need any of the `stuck.pack` stuff? Is that just
+        # padding the messages for TCP?
+        # I've included it now for unpacking.
+        lengths = [nbytes(frame) for frame in frames]
+        length_bytes = ([struct.pack('Q', len(frames))] +
+                        [struct.pack('Q', x) for x in lengths])
+        b = b''.join(length_bytes + frames)  # small enough, send in one go
```

Comment:
Mostly I'm trying to avoid this memory copy and the coercion of memoryviews into bytes objects